packages/api/src/controllers/signing-key.ts

```diff
@@ -0,0 +1,238 @@
+};
+
+function generateSigningKeys() {
+  const keypair = generateKeyPairSync("rsa", {
+    modulusLength: 4096,
+    publicKeyEncoding: {
+      type: "spki",
+      format: "pem",
+    },
+    privateKeyEncoding: {
+      type: "pkcs8",
+      format: "pem",
+    },
+  });
+  return keypair;
+}
+
+const app = Router();
+
+app.get("/", authorizer({}), async (req, res) => {
+  let { limit, cursor, all, allUsers, order, filters, count } = toStringValues(
+    req.query
+  );
+  if (isNaN(parseInt(limit))) {
+    limit = undefined;
+  }
+  if (!order) {
+    order = "updatedAt-true,createdAt-true";
+  }
+
+  if (req.user.admin && allUsers && allUsers !== "false") {
+    const query = parseFilters(fieldsMap, filters);
+    if (!all || all === "false") {
+      query.push(sql`signing_key.data->>'deleted' IS NULL`);
+    }
+
+    let fields =
+      " signing_key.id as id, signing_key.data as data, users.id as usersId, users.data as usersdata";
+    if (count) {
+      fields = fields + ", count(*) OVER() AS count";
+    }
+    const from = `signing_key left join users on signing_key.data->>'userId' = users.id`;
+    const [output, newCursor] = await db.signingKey.find(query, {
+      limit,
+      cursor,
+      fields,
+      from,
+      order: parseOrder(fieldsMap, order),
+      process: ({ data, usersdata, count: c }) => {
+        if (count) {
+          res.set("X-Total-Count", c);
+        }
+        return {
+          data,
+          user: db.user.cleanWriteOnlyResponse(usersdata),
+        };
+      },
+    });
+
+    res.status(200);
+
+    if (output.length > 0 && newCursor) {
+      res.links({ next: makeNextHREF(req, newCursor) });
+    }
+    return res.json(output);
+  }
+
+  const query = parseFilters(fieldsMap, filters);
+  query.push(sql`signing_key.data->>'userId' = ${req.user.id}`);
+  query.push(sql`signing_key.data->>'deleted' IS NULL`);
+
+  let fields = " signing_key.id as id, signing_key.data as data";
+  if (count) {
+    fields = fields + ", count(*) OVER() AS count";
+  }
+  const from = `signing_key`;
+  const [output, newCursor] = await db.signingKey.find(query, {
+    limit,
+    cursor,
+    fields,
+    from,
+    order: parseOrder(fieldsMap, order),
+    process: ({ data, count: c }) => {
+      if (count) {
+        res.set("X-Total-Count", c);
+      }
+      return data;
+    },
+  });
+
+  res.status(200);
+
+  if (output.length > 0) {
+    res.links({ next: makeNextHREF(req, newCursor) });
+  }
+
+  return res.json(output);
+});
+
+app.get("/:id", authorizer({}), async (req, res) => {
+  const signingKey = await db.signingKey.get(req.params.id);
+  if (!signingKey || signingKey.deleted) {
+    res.status(404);
+    return res.json({
+      errors: ["not found"],
+    });
+  }
+
+  if (req.user.admin !== true && req.user.id !== signingKey.userId) {
+    res.status(403);
+    return res.json({
+      errors: ["user can only request information on their own signingKeys"],
```

Comment:
WDYT making this just a `404` as well?

I have mixed feelings, and we are not consistent across all APIs, but since signing-keys are somewhat more sensitive data maybe we want to hide when it exists but belongs to another user.

Reply:
Sounds good!